lib/App/git/ship/perl.pm

```diff
@@ -139,7 +139,7 @@ sub ship {
   $self->system(qw( git add Makefile.PL ), $changelog, $self->_filename('readme'));
   $self->system(qw( git commit -a -m ), $self->_changes_to_commit_message);
   $self->SUPER::ship(@_);    # after all the changes
-  $uploader->upload_file($dist_file);
+  $uploader->upload_file($dist_file) unless $ENV{GIT_SHIP_NO_UPLOAD};
```

Comment:
The logic will be if GIT_SHIP_UPLOAD=0 then don't upload, but it defaults to 1 unless defined. Anyway, I don't mind having GIT_SHIP_NO_UPLOAD. Maybe it should be something you can set in .ship.conf as well?